sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -713,7 +713,10 @@ fixed topology that can be submitted for execution on any queue sharing the
 context associated with the graph. It is valid to call this method multiple times
 to create subsequent executable graphs. It is also valid to continue to add new
 nodes to the modifiable graph instance after calling this function. It is valid
-to finalize an empty graph instance with no recorded commands.
+to finalize an empty graph instance with no recorded commands. The created
+executable graph(s) from the same non-empty modifiable graph could not be executed
+simultaneously because they use the same memory address recorded in the modifiable
```

Comment:
I think "use" here is too general. Is this UB not only an issue if one of the graphs is doing a write to the memory address? If the graphs were only reading memory would this be an issue? (I know this is contrived) 
Prefer something more specific, like
> should not be executed simultaneously if at least one executable graph contains a write operation to a memory address accessed by another executable graph

Reply:
1. If a modifiable graph is modified after finalization, is it considered to be the same or not?

2. Documentation for `handler::ext_oneapi_graph` says "Concurrent graph execution can be achieved by finalizing a graph in modifiable state into multiple graphs in executable state." Which, on the surface, contradicts this statement here (at least in the common case of two copies of the graph writing to the same memory location)?

Reply:
> 1. If a modifiable graph is modified after finalization, is it considered to be the same or not?
It depends if a same memory write is in the executions, to be accurate, I'll remove "from the same non-empty modifiable graph"

> 2. Documentation for `handler::ext_oneapi_graph` says "Concurrent graph execution can be achieved by finalizing a graph in modifiable state into multiple graphs in executable state." Which, on the surface, contradicts this statement here (at least in the common case of two copies of the graph writing to the same memory location)?

nice catch, I think the concurrent graph execution is not allowed for most cases, wait for more comments

Reply:
> nice catch, I think the concurrent graph execution is not allowed for most cases, wait for more comments

And to clarify, the problem is only with execution, not submission? It is legal to submit two executable graphs created from the same modifiable graph are submitted into an in-order queue one after the other, right?

Reply:
> > nice catch, I think the concurrent graph execution is not allowed for most cases, wait for more comments
> 
> And to clarify, the problem is only with execution, not submission? It is legal to submit two executable graphs created from the same modifiable graph are submitted into an in-order queue one after the other, right?

yes, I think so.  It is legal to submit the  executable graph(s) again and again into an in-order queue, there will be no race condition for the device memory.

Reply:
Something to keep in mind for this PR is that in future a graph update API could modify the arguments used by an executable graph. So we want the wording about UB to be defined in terms of the race conditions accessing the memory used by concurrent executable graphs, rather than focusing on the modifiable graph they were created from. I think the current wording achieves that though 👍 

> And to clarify, the problem is only with execution, not submission? It is legal to submit two executable graphs created from the same modifiable graph are submitted into an in-order queue one after the other, right?

That is correct, it would also be valid to submit the executable graph(s) to an out-of-order queue with a dependency between the submissions.

> Documentation for handler::ext_oneapi_graph says "Concurrent graph execution can be achieved by finalizing a graph in modifiable state into multiple graphs in executable state." Which, on the surface, contradicts this statement here (at least in the common case of two copies of the graph writing to the same memory location)?

You are correct, good spot. I'd suggest removing this sentence as part of this PR.